pkg/results/TablePrinter.go

```diff
@@ -697,12 +709,35 @@ func (t *TextTablePrinter) Footer(ctx context.Context, result *Result) error {
 		hitsTotal = strings.TrimSpace(textFormatter.Count(uint64(result.Summary.Hits.Total)))
 	}
 
+	result.Query.PrintFooter(t.footerWriter)
 	t.footerWriter.WriteEntry(queryStatsKey, "displayed top %s hits out of %s in %s",
 		hitsDisplayed,
 		hitsTotal,
 		textFormatter.Duration(result.Summary.Timings.QueryDuration),
 	)
-	result.Query.PrintFooter(t.footerWriter)
+
+	if t.printQueryStats {
+		stats := result.Summary.Stats
```

Comment:
Can this be nil?

Reply:
It can't. Otherwise, we'd have a panic way earlier :). Also, check `results.New()`, which will initialize the stats.